sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp

```diff
@@ -121,91 +143,532 @@ struct SptrsvWrap {
           lhs(lhs_),
           rhs(rhs_),
           nodes_grouped_by_level(nodes_grouped_by_level_) {
-      KK_REQUIRE_MSG(!BlockEnabled, "Blocks are not yet supported.");
-      KK_REQUIRE_MSG(block_size_ == 0, "Blocks are not yet supported.");
+      KK_REQUIRE_MSG(block_size_ == 0,
+                     "Tried to use blocks with the unblocked Common?");
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    size_type get_block_size() const { return 0; }
+
+    // lset
+    KOKKOS_INLINE_FUNCTION
+    void lset(const size_type row, const scalar_t value) const {
+      lhs(row) = value;
+    }
+
+    // add. y += x
+    KOKKOS_INLINE_FUNCTION
+    static void add(const member_type &team, const scalar_t &x, scalar_t &y) {
+      Kokkos::single(Kokkos::PerTeam(team), [&]() { y += x; });
+      team.team_barrier();
+    }
+
+    // serial add. y += x
+    KOKKOS_INLINE_FUNCTION
+    static void add(const scalar_t &x, scalar_t &y) { y += x; }
+
+    // divide. b /= A
+    KOKKOS_INLINE_FUNCTION
+    static void divide(const member_type &team, scalar_t &b, const scalar_t &A,
+                       scalar_t *) {
+      Kokkos::single(Kokkos::PerTeam(team), [&]() { b /= A; });
+      team.team_barrier();
+    }
+
+    // serial divide. b /= A
+    KOKKOS_INLINE_FUNCTION
+    static void divide(scalar_t &b, const scalar_t &A, scalar_t *) { b /= A; }
+
+    // multiply_subtract. C -= A * B
+    KOKKOS_INLINE_FUNCTION
+    static void multiply_subtract(const scalar_t &A, const scalar_t &B,
+                                  scalar_t &C) {
+      C -= A * B;
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    static void copy(const member_type &, scalar_t &, const scalar_t &) {}
```

Comment:
It is a bit strange that copy does not actually perform a copy? Is it okay for the two scalar to potentially have different values after a call to this function?
I would rather that we drop this overload if possible.

Reply:
The copy is to get the reduce item (ArrayType) values back into `lhs_val` for the blocked case. In the unblocked case, we reduce directly into lhs_val, so nothing needs to be done, but the `copy` function still needs to exist or else the code won't compile. I could protect the `copy` calls with an `if constexpr` if you'd like, that would allow me to delete the functions from the unblocked `Common` impl.